### PR TITLE
Instrument metrics reporting to Redis

### PR DESCRIPTION
**Pull request by lotif**
# PR Type
Feature

# Short Description

Clickup Ticket(s): https://app.clickup.com/t/8687c8jj1

Adding a subclass of `MetricsReporter` that saves the metrics to Redis instead of a file.
Tested by starting a server with it:
```python
metrics_reporter = RedisMetricsReporter(redis_connection=redis.Redis(host="localhost", port=6379))
server = FlServer(client_manager, strategy, metrics_reporter=metrics_reporter)
```
After training, check if the metrics have been saved to Redis successfully:
```shell
$ python
Python 3.9.6 (default, Nov 10 2023, 13:38:27) 
[Clang 15.0.0 (clang-1500.1.0.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import redis
>>> server_conn = redis.Redis(host="localhost", port=6379)
>>> server_conn.get("da37a3fe-7c68-44a7-98bb-817952212a43")
b'{"type": "server", "fit_start": "2024-02-27 12:11:00.913371",  ...
```

# Tests Added
Unit tests for RedisMetricsReporter.

### CODE REVIEW

File: .github/workflows/integration_tests.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: tests
+name: integration-tests
```
Comment (jewelltaylor) [inline .github/workflows/integration_tests.yaml:1]: sorry i should have done this! didnt think to split them up

File: florist/api/monitoring/metrics.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import json
+from logging import DEBUG
+from typing import Any, Dict, Optional
+
+import redis
+from fl4health.reporting.metrics import DateTimeEncoder, MetricsReporter
+from flwr.common.logger import log
+
+
+class RedisMetricsReporter(MetricsReporter):  # type: ignore
```
Comment (jewelltaylor) [inline florist/api/monitoring/metrics.py:11]: curious why we have to type ignore this

Comment (jewelltaylor) [reply florist/api/monitoring/metrics.py:11]: ahhh no output folder

File: florist/api/monitoring/metrics.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    def __init__(
+        self,
+        redis_connection: redis.client.Redis,
+        run_id: Optional[str] = None,
+    ):
+        """
+        Init an instance of RedisMetricsReporter.
+
+        :param redis_connection: (redis.client.Redis) the connection object to a Redis. Should be the output
```
Comment (jewelltaylor) [inline florist/api/monitoring/metrics.py:22]: I noticed you are using sphinx docstring format and I used the google format in the last PR. I don't have a strong preference and would be happy to adopt sphinx, do you think we should use the same format? Or does it not matter

**Review by jewelltaylor — APPROVED** [review]
Only one comment to clarify if we should stay consistent on docstring format. Other than that, everything looks great and runs on my local machine

File: .github/workflows/integration_tests.yaml (hunk 1 above)
Comment (lotif) [reply .github/workflows/integration_tests.yaml:1]: No problems! Things were still in flux with my open PR on aieng-template changes.

File: florist/api/monitoring/metrics.py (hunk 3 above)
Comment (lotif) [reply florist/api/monitoring/metrics.py:22]: I just finished setting up automatic documentation with sphinx (following aieng-template changes) so I think we should stick to the sphinx format.

https://vectorinstitute.github.io/FLorist/reference/api/florist.api.launchers.launch.html

File: florist/api/monitoring/metrics.py (hunk 2 above)
Comment (lotif) [reply florist/api/monitoring/metrics.py:11]: It was this weird error:
https://stackoverflow.com/questions/49888155/class-cannot-subclass-qobject-has-type-any-using-mypy

I tried solving it a couple of different ways without luck, so I ignored it. Seems to be something mypy is not picking up from fl4health's package settings and not really a big issue. Not sure if it's worth fixing it in fl4health. We can also disable this check if it becomes too annoying.

File: florist/api/monitoring/metrics.py (hunk 3 above)
Comment (jewelltaylor) [reply florist/api/monitoring/metrics.py:22]: Ahhhhh I seee, makes sense! I will keep that in mind going forward

**Pull request merged**